Add unit test cases for vfio-ioctls (#44)
rust-vmm/vfio-ioctls: crates/vfio-ioctls/README.md

# vfio-ioctls

## Design

The [VFIO driver framework](https://www.kernel.org/doc/Documentation/vfio.txt)
provides unified APIs for direct device access. It is an IOMMU/device-agnostic framework for
exposing direct device access to user space in a secure, IOMMU-protected environment.
This framework is used for multiple devices, such as GPUs, network adapters, and compute
accelerators. With direct device access, virtual machines or user space applications have
direct access to the physical device.

> The VFIO framework is originally developed on Linux system, and later Microsoft HyperV

Reviewer: `Microsoft HyperV` should be `Microsoft Hypervisor`

> technology provides an compatible implementation. So the VFIO framework is supported

Reviewer 2: ```suggestion
technology provides a compatible implementation. Therefore the VFIO framework is supported
```

> by both Linux and Windows HyperV.

Reviewer: Same here.

The `vfio-ioctls` crate is a safe wrapper over the VFIO APIs. It provides three classes of structs:
- `VfioContainer`: a safe wrapper over a VFIO container object, and acts a container object
  to associate `VfioDevice` objects with IOMMU domains.
- `VfioDevice`: a wrapper over a VFIO device object, provide methods to access the underlying
  hardware device.
- `VfioIrq/VfioRegion`: describes capabilities/resources about a `VfioDevice` object. 

## Usage

The `vfio-ioctls` crate may be used to support following usage scenarios:
- Direct device assignment to virtual machine based on Linux KVM, with default features.

> - Direct device assignment to virtual machine based on Microsoft HyperV, with `--no-default-features --features=mshv`.

Reviewer: Same here.
